Src/Newtonsoft.Json/Utilities/EnumUtils.cs

```diff
@@ -231,16 +229,16 @@ public static object ParseEnumName(string enumText, bool isNullable, bool disall
             return Enum.Parse(t, finalEnumText, true);
```

Comment:
If you change the third parameter and change it to false for Enum.Parse you will also support enum members which only differs in case.

See testcase DeserializeEnumWithDifferentCases

Since the EnumBidirectionalDictionary now support case insentitivity it will automatically match the required case since it loads the enumMember from the EnumBidirectionalDictionary.

2 other test cases can be seen here.
`
        [JsonConverter(typeof(StringEnumCaseInsensitiveConverter))]
        public enum EnumMemberWithDiffrentCases
        {
            [EnumMember(Value = "first_value")]
            First,
            [EnumMember(Value = "First_value")]
            first
        }

        [Test]
        public void DeserializeEnumMemberWithDifferentCasing_ByEnumMemberValue_First()
        {
            var e = JsonConvert.DeserializeObject<EnumMemberWithDiffrentCases>(@"""first_value""", new StringEnumCaseInsensitiveConverter());
            Assert.AreEqual(EnumMemberWithDiffrentCases.First, e);
        }

        [Test]
        public void DeserializeEnumMemberWithDifferentCasing_ByEnumMemberValue_Second()
        {
            var e = JsonConvert.DeserializeObject<EnumMemberWithDiffrentCases>(@"""First_value""", new StringEnumCaseInsensitiveConverter());
            Assert.AreEqual(EnumMemberWithDiffrentCases.first, e);
        }
`

Reply:
The third parameter should match the case-sensitivity requested by the developer (instead of being hard-coded to a given Boolean value). This would empower developers to chose which behavior they want and would also ensure the case sensitivity when converting by EnumMember value and when converting by enum item name are consistent.

However, there are 3 existing unit tests that assume `Enum.Parse` is always case insensitive despite the fact they use the case sensitive converter (which is contradictory, IMHO).

The first unit test:

```csharp
[Test]
public void DeserializeFlagEnum()
{
    string json = @"{
        ""StoreColor"": ""Red, White"",
        ""NullableStoreColor1"": 0,
        ""NullableStoreColor2"": ""black, Red, White""
    }";

    EnumClass enumClass = JsonConvert.DeserializeObject<EnumClass>(json, new StringEnumConverter());

    Assert.AreEqual(StoreColor.Red | StoreColor.White, enumClass.StoreColor);
    Assert.AreEqual((StoreColor)0, enumClass.NullableStoreColor1);
    Assert.AreEqual(StoreColor.Red | StoreColor.White | StoreColor.Black, enumClass.NullableStoreColor2);
}
```

In this example, notice that "black" is converted to `StoreColor.Black` despite the fact that a case sensitive converter is used. This is a trivial situation that can be resolved by either capitalizing the "b" or by changing to a case insensitive converter.

The other two cases are a little more complex: they involve camel casing.

```csharp
[Test]
public void CamelCaseTextFlagEnumDeserialization()
{
    string json = @"{
        ""Enum"": ""first, second""
    }";

    EnumContainer<FlagsTestEnum> c = JsonConvert.DeserializeObject<EnumContainer<FlagsTestEnum>>(json, new StringEnumConverter { CamelCaseText = true });
    Assert.AreEqual(FlagsTestEnum.First | FlagsTestEnum.Second, c.Enum);
}

[Test]
public void Deserialize_CamelCaseFromAttribute()
{
    CamelCaseEnum e = JsonConvert.DeserializeObject<CamelCaseEnum>(@"""camelCase""");
    Assert.AreEqual(CamelCaseEnum.CamelCase, e);
}
```

Notice that "first" is converted to `FlagsTestEnum.First`, "second" is converted to `FlagsTestEnum.Second` and "camelCase" is converted to `CamelCaseEnum.CamelCase`.

I'm not sure how to resolve this situation. It doesn't make sense to desire a converter that is both case sensitive **AND** allows camel cased values at the same time.